File: LR4/src/main/java/org/example/comparator/IntegerComparator.java
package org.example.comparator;

import org.example.types.IntegerType;

import java.io.Serializable;

public class IntegerComparator implements Comparator, Serializable {
    @Override
    public int compare(Object o1, Object o2) {
        return ((IntegerType) o1).getIntegerValue() - ((IntegerType) o2).getIntegerValue();
    }
}

File: LR2/src/main/kotlin/comparator/IntegerComparator.kt
package comparator

import types.IntegerType
import java.io.Serializable

class IntegerComparator : Comparator, Serializable {
    override fun compare(o1: Any, o2: Any): Int {
        return (o1 as IntegerType).getIntegerValue() - (o2 as IntegerType).getIntegerValue()
    }
}
